Release notes for CI updates phase 2 (#4051)
dbt-labs/docs.getdbt.com: website/docs/docs/dbt-versions/release-notes/04-Sept-2023/ci-updates-phase2-rn.md

---
title: "Update: Improvements to dbt Cloud continuous integration"
description: "September 2023: dbt Cloud now has two types of jobs &mdash; deploy jobs and CI jobs &mdash; with streamlined setup and improved efficiency. "
sidebar_label: "Update: Improvements to dbt jobs"
tags: [Sept-2023, CI]
date: 2023-09-11
sidebar_position: 10
---

> dbt Labs have separated out [deploy jobs](/docs/deploy/deploy-jobs), used to build production data assets, and [CI jobs](/docs/deploy/ci-jobs), used to check code changes, into two distinct types of jobs. These jobs perform fundamentally different tasks so we’ve improved the setup experience with sensible defaults for each. 

Author: ```suggestion
dbt Cloud now separates [deploy jobs](/docs/deploy/deploy-jobs), used to build production data assets, and [CI jobs](/docs/deploy/ci-jobs), used to check code changes, into two distinct types of jobs. These jobs perform fundamentally different tasks so dbt Labs have improved the setup experience with sensible defaults for each. 
```

> dbt Labs have separated out [deploy jobs](/docs/deploy/deploy-jobs), used to build production data assets, and [CI jobs](/docs/deploy/ci-jobs), used to check code changes, into two distinct types of jobs. These jobs perform fundamentally different tasks, and we’ve been improving the experience of both for you to have better fit for purpose tools. 

Reviewer: ```suggestion
dbt Cloud now separates [deploy jobs](/docs/deploy/deploy-jobs), used to build production data assets, and [CI jobs](/docs/deploy/ci-jobs), used to check code changes, into two distinct types of jobs. These jobs perform fundamentally different tasks, so we’ve created a streamlined setup experience with sensible defaults for each.
```

> dbt Labs have separated out [deploy jobs](/docs/deploy/deploy-jobs), used to build production data assets, and [CI jobs](/docs/deploy/ci-jobs), used to check code changes, into two distinct types of jobs. These jobs perform fundamentally different tasks, and we’ve been improving the experience of both for you to have better fit for purpose tools. 

Author: ```suggestion
dbt Labs have separated out [deploy jobs](/docs/deploy/deploy-jobs), used to build production data assets, and [CI jobs](/docs/deploy/ci-jobs), used to check code changes, into two distinct types of jobs. These jobs perform fundamentally different tasks so we’ve improved the setup experience with sensible defaults for each. 
```

Since there are now two types of jobs, instead of one generic type, we can better guide you through the setup flow. Best practices are built into the default settings so you can go from curious to being set up in seconds.

And, we now have more efficient state comparisons on CI checks: Never waste a build or test on code that hasn’t been changed. We now diff between the PR code and what’s running in production more efficiently, with the introduction of deferral to an environment versus a job. To learn more, refer to [Continuous integration in dbt Cloud](/docs/deploy/continuous-integration).

Below is a comparison table that describes how deploy jobs and CI jobs behave differently:

|  | Deploy Jobs | CI Jobs |
| --- | --- | --- |
| Purpose | Builds production data assets. | Builds and tests new code before merging changes into production. |

> | Trigger types | Triggered by a schedule or by API. | Triggered by a webhook from a commit to a PR or by API. |

Reviewer: ```suggestion
| Trigger types | Triggered by a schedule or by API. | Triggered by a commit to a PR or by API. |
```
It's technically true that webhooks make this happen behind the scenes, but it's not dbt Cloud's webhooks, it's the git provider's webhook. So talking about webhooks here is a bit confusing ("do I need to set up a dbt Cloud webhook to make this work?" no)

| Destination | Builds into a production database and schema. | Builds into a staging database and ephemeral schema, lived for the lifetime of the PR. |
| Execution mode | Runs execute sequentially, so as to not have collisions on the underlying DAG. | Runs execute in parallel to promote team velocity. |

> | Efficiency run savings | Detects over-scheduled jobs and cancels unnecessary runs to avoid queue clog. | Cancels runs when an in-flight run becomes stale when a new commit is pushed to the pull request. |

Reviewer: ```suggestion
| Efficiency run savings | Detects over-scheduled jobs and cancels unnecessary runs to avoid queue clog. | Cancels existing runs when a newer commit is pushed to avoid redundant work. |
```

---
title: "Update: Improvements to dbt Cloud continuous integration"
description: "September 2023: dbt Cloud now has two types of jobs &mdash; deploy jobs and CI jobs &mdash; with streamlined setup and improved efficiency. "
sidebar_label: "Update: Improvements to dbt jobs"
tags: [Sept-2023, CI]
date: 2023-09-11
sidebar_position: 10
---

> dbt Cloud now separates [deploy jobs](/docs/deploy/deploy-jobs), used to build production data assets, and [CI jobs](/docs/deploy/ci-jobs), used to check code changes, into two distinct types of jobs. These jobs perform fundamentally different tasks so dbt Labs have improved the setup experience with sensible defaults for each. 

Author: ```suggestion
dbt Cloud now has two distinct job types: [deploy jobs](/docs/deploy/deploy-jobs) for building production data assets, and [CI jobs](/docs/deploy/ci-jobs) for checking code changes. These jobs perform fundamentally different tasks so dbt Labs improved the setup experience with better defaults for each. 
```

> Since there are now two types of jobs, instead of one generic type, we can better guide you through the setup flow. Best practices are built into the default settings so you can go from curious to being set up in seconds.

Author: ```suggestion
With two types of jobs, instead of one generic type, we can better guide you through the setup flow. Best practices are built into the default settings so you can go from curious to being set up in seconds.
```

<Lightbox src="/img/docs/release-notes/ci-job-setup.gif" title="Example of setting up a CI job"/>

> And, we now have more efficient state comparisons on CI checks: Never waste a build or test on code that hasn’t been changed. We now diff between the PR code and what’s running in production more efficiently, with the introduction of deferral to an environment versus a job. To learn more, refer to [Continuous integration in dbt Cloud](/docs/deploy/continuous-integration).

Author: ```suggestion
And, we now have more efficient state comparisons on CI checks: never waste a build or test on code that hasn’t been changed. We now diff between the Git PR code and what’s running in production more efficiently, with the introduction of deferral to an environment versus a job. To learn more, refer to [Continuous integration in dbt Cloud](/docs/deploy/continuous-integration).
```

> And, we now have more efficient state comparisons on CI checks: never waste a build or test on code that hasn’t been changed. We now diff between the Git PR code and what’s running in production more efficiently, with the introduction of deferral to an environment versus a job. To learn more, refer to [Continuous integration in dbt Cloud](/docs/deploy/continuous-integration).

Author: ```suggestion
And, we now have more efficient state comparisons on CI checks: never waste a build or test on code that hasn’t been changed. We now diff between the Git PR code and what’s running in production more efficiently, with the introduction of deferral to an environment versus a job. To learn more, refer to [Continuous integration in dbt Cloud](/docs/deploy/continuous-integration) and [Get started with continuous integration tests](/guides/orchestration/set-up-ci/overview).
```

> And, we now have more efficient state comparisons on CI checks: never waste a build or test on code that hasn’t been changed. We now diff between the Git PR code and what’s running in production more efficiently, with the introduction of deferral to an environment versus a job. To learn more, refer to [Continuous integration in dbt Cloud](/docs/deploy/continuous-integration) and [Get started with continuous integration tests](/guides/orchestration/set-up-ci/overview).

Author: ```suggestion
And, we now have more efficient state comparisons on CI checks: never waste a build or test on code that hasn’t been changed. We now diff between the Git PR code and what’s running in production more efficiently with the introduction of deferral to an environment versus a job. To learn more, refer to [Continuous integration in dbt Cloud](/docs/deploy/continuous-integration) and [Get started with continuous integration tests](/guides/orchestration/set-up-ci/overview).
```
